homeworks/konstantin.tsarenko_KTsarenko/homework_6/hw-6.js

```diff
@@ -0,0 +1,144 @@
+
+// task 1
+
+const obj = {
+
+    get fullName() {
+        return this.valueFullName;
+    },
+
+    set fullName(value) {
+        this.valueFullName = value.toLowerCase().replace(/\b\w/g, l => l.toUpperCase());
+    },
+
+    get phone() {
+        return this.valuePhone;
+    },
+
+    set phone(value) {
+        this.valuePhone = value.match(/^\+|\d/g).join('');
+    },
+};
+
+obj.fullName = 'aNna-mAria joHNs';
+console.log(obj.fullName);
+
+obj.phone = '‎‎+380(067)440-20-57';
+console.log(obj.phone);
+
+// task 2
+
+class Horse {
+    constructor(name, breed) {
+        this.name = name;
+        this.breed = breed;
+    }
+}
+
+class Racer extends Horse {
+    constructor(name, breed) {
+        super(name, breed);
+        this.distance = 0;
+        this.speed = this.setSpeed();
+    }
+
+    setSpeed() {
+        this.min = 10;
+        this.max = 16;
+        return Math.floor(Math.random() * (this.max - this.min)) + this.min;
+    }
+
+    run() {
+        return new Promise((resolve) => {
+            const race = setInterval(() => {
+                this.distance += this.speed;
+                this.speed = this.setSpeed();
+            }, 1000);
+            const timerRun = setInterval(() => {
+                console.log(`name: ${this.name} breed: ${this.breed} distance: ${this.distance} `);
+            }, 2000);
+            setTimeout(() => {
+                clearInterval(timerRun);
+                clearInterval(race);
+                resolve();
+            }, 10000);
```

Comment:
`run()` should not include race related functionality, logically it's wrong, `run()` should only make horse run

Reply:
removed